src/vs/workbench/browser/web.main.ts

```diff
@@ -358,6 +359,20 @@ export class BrowserMain extends Disposable {
 		const userDataInitializationService = new UserDataInitializationService(environmentService, credentialsService, userDataSyncStoreManagementService, fileService, userDataProfilesService, storageService, productService, requestService, logService, uriIdentityService);
 		serviceCollection.set(IUserDataInitializationService, userDataInitializationService);
 
+		try {
+			await Promise.race([
+				// Do not block more than 5s
+				timeout(5000),
```

Comment:
@sandy081, would it make sense to make this configurable so that the embedder can adjust it without code changes on our side?

Reply:
I do not think so as it is hard for the embedder or user to configure any good value for it. It is just a decision that workbench takes not to block.